### PR TITLE
errbot module: init

### PR DESCRIPTION
**Pull request by globin**

### CODE REVIEW

**Review by LnL7 — APPROVED** [review]

File: nixos/modules/services/misc/errbot.nix [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  '';
+in {
+  options = {
+    services.errbot.instances = mkOption {
+      default = {};
+      description = "Errbot instance configs";
+      type = types.attrsOf (types.submodule {
+        options = {
+          dataDir = mkOption {
+            type = types.nullOr types.str;
```
Comment (joachifm) [inline nixos/modules/services/misc/errbot.nix]: I think `types.path` makes sense here

File: nixos/modules/services/misc/errbot.nix [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            default = null;
+            description = "Data directory for errbot instance.";
+          };
+
+          plugins = mkOption {
+            type = types.listOf types.path;
```
Comment (joachifm) [inline nixos/modules/services/misc/errbot.nix]: I'm guessing `types.package` was meant here?

File: nixos/modules/services/misc/errbot.nix [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      preStart = ''
+        mkdir -p ${dataDir}
+        chown -R errbot:errbot ${dataDir}
+      '';
+      serviceConfig = {
+        Type = "simple";
+        User = "errbot";
+        Group = "errbot";
+        Restart = "on-failure";
+        ExecStart="${pkgs.errbot}/bin/errbot -c ${mkConfigFile instanceCfg dataDir}";
```
Comment (joachifm) [inline nixos/modules/services/misc/errbot.nix]: Nit: spaces around `=`

File: nixos/modules/services/misc/errbot.nix [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      preStart = ''
+        mkdir -p ${dataDir}
+        chown -R errbot:errbot ${dataDir}
+      '';
+      serviceConfig = {
+        Type = "simple";
+        User = "errbot";
+        Group = "errbot";
```
Comment (joachifm) [inline nixos/modules/services/misc/errbot.nix]: Nit: strictly redundant.

File: nixos/modules/services/misc/errbot.nix [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        "/var/lib/errbot/${name}";
+    in {
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      preStart = ''
+        mkdir -p ${dataDir}
+        chown -R errbot:errbot ${dataDir}
+      '';
+      serviceConfig = {
+        Type = "simple";
```
Comment (joachifm) [inline nixos/modules/services/misc/errbot.nix]: Nit: strictly redundant.

**Comment by globin** [comment]
Updated

**Pull request merged**